File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/SetParkingLotsPresenter.java
package com.mobile.tandil.javabaseproject.mvp.presenter;

import com.mobile.tandil.javabaseproject.listeners.ParkingLotsInputListener;
import com.mobile.tandil.javabaseproject.mvp.contract.SetParkingLotsContract;

public class SetParkingLotsPresenter implements SetParkingLotsContract.SetParkingLotsPresenter {
    private final SetParkingLotsContract.SetParkingLotsView view;
    private final ParkingLotsInputListener parkingLotsInputListener;

    public SetParkingLotsPresenter(SetParkingLotsContract.SetParkingLotsView view, ParkingLotsInputListener parkingLotsInputListener) {
        this.view = view;
        this.parkingLotsInputListener = parkingLotsInputListener;
    }

    @Override
    public void onSubmitButtonPressed() {
        String lotsString = view.getParkingLots();
        if (lotsString.isEmpty()) {
            view.showErrorMessage();
        } else {
            int lots = Integer.valueOf(lotsString);
            this.parkingLotsInputListener.shareLotsInput(lots);
            view.showParkingLots(lots);
            view.closeSetParkingLotsDialog();
        }
    }

    @Override
    public void onCancelButtonPressed() {
        view.closeSetParkingLotsDialog();
    }
}

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/SetParkingLotsPresenter.kt
package com.mobile.tandil.javabaseproject.mvp.presenter

import com.mobile.tandil.javabaseproject.databinding.DialogFragmentSetParkingLotsBinding
import com.mobile.tandil.javabaseproject.mvp.contract.SetParkingLotsContract

class SetParkingLotsPresenter(private val view: SetParkingLotsContract.View) : SetParkingLotsContract.Presenter {

    init {
        view.onSubmitBtnClick { onSubmitButtonPressed() }
        view.onCancelBtnClick { onCancelButtonPressed() }
    }

    override fun onSubmitButtonPressed() {
        val lotsString: String = view.getParkingLots()
        if (lotsString.isEmpty()) {
            view.showErrorMessage()
        } else {
            val lots = lotsString.toInt()
            view.showParkingLots(lots)
            view.closeSetParkingLotsDialog()
        }
    }

    override fun onCancelButtonPressed() {
        view.closeSetParkingLotsDialog()
    }

    override fun getBinding(): DialogFragmentSetParkingLotsBinding {
        return view.getBinding()
    }
}
